frameworks/go/codewars/reporter/reporter.go

```diff
@@ -0,0 +1,87 @@
+package reporter
+
+import (
+  "github.com/onsi/ginkgo/config"
+  "github.com/onsi/ginkgo/types"
+  "fmt"
+  "strings"
+)
+
+// uses map to keep track of nested describes, good enough for now
+type cwReporter struct {
+  specs map[int]string
+}
+
+func New() *cwReporter {
+  return &cwReporter{
+    specs: make(map[int]string),
+  }
+}
+
+func (r *cwReporter) SpecSuiteWillBegin(config config.GinkgoConfigType, summary *types.SuiteSummary) {
+  fmt.Printf("<DESCRIBE::>%s\n", escape(summary.SuiteDescription))
+}
+
+func (r *cwReporter) BeforeSuiteDidRun(setupSummary *types.SetupSummary) {
+}
+
+func (r *cwReporter) AfterSuiteDidRun(setupSummary *types.SetupSummary) {
+}
+
+func (r *cwReporter) SpecWillRun(spec *types.SpecSummary) {
+  n := len(spec.ComponentTexts)
+  for i := 1; i + 1 < n; i++ {
+    v := spec.ComponentTexts[i]
+    // append LineNumer to support multiple test cases with same name
+    id := fmt.Sprintf("%s:%d", v, spec.ComponentCodeLocations[i].LineNumber)
+    l, ok := r.specs[i]
+    if !ok || l != id {
+      m := len(r.specs)
+      if ok {
+        for j := i; j <= m; j++ {
+          fmt.Println("<COMPLETEDIN::>")
+          delete(r.specs, j)
+        }
+      }
+      r.specs[i] = id
+      fmt.Printf("<DESCRIBE::>%s\n", escape(v))
+    }
+  }
+
+  fmt.Printf("<IT::>%s\n", escape(spec.ComponentTexts[n - 1]))
+}
+
+func (r *cwReporter) SpecDidComplete(spec *types.SpecSummary) {
+  switch spec.State {
+    case types.SpecStatePassed:
+      fmt.Println("<PASSED::>Test Passed")
+    case types.SpecStateFailed:
+      fmt.Println("<FAILED::>Test Failed")
+      fmt.Printf("%s\n", escape(spec.Failure.Message))
+    case types.SpecStatePanicked:
+      fmt.Println("<ERROR::>")
+    case types.SpecStateTimedOut:
+    case types.SpecStateSkipped:
+    case types.SpecStatePending:
+  }
+  fmt.Printf("<COMPLETEDIN::>%.4f\n", spec.RunTime.Seconds() * 1000)
+}
+
+func (r *cwReporter) SpecSuiteDidEnd(summary *types.SuiteSummary) {
+  m := len(r.specs)
+  for j := 1; j <= m; j++ {
+    fmt.Println("<COMPLETEDIN::>")
+  }
+  fmt.Printf("Passed: %d Failed: %d\n", summary.NumberOfPassedSpecs, summary.NumberOfFailedSpecs)
+  fmt.Printf("<COMPLETEDIN::>%.4f\n", summary.RunTime.Seconds() * 1000)
+}
+
+func escape(s string) string {
+  s = strings.Replace(s, "&", "&amp;", -1)
+  s = strings.Replace(s, "\"", "&quot;", -1)
+  s = strings.Replace(s, "'", "&#39;", -1)
+  s = strings.Replace(s, "<", "&lt;", -1)
```

Comment:
Thanks for reviewing! I'll remove those lines.

---

For the version, I will try 1.7 for now because 1.8 isn't released yet. Maybe in few weeks I can upgrade to 1.8.
> Go 1.8 is not yet released. These are work-in-progress release notes. Go 1.8 is expected to be released in February 2017.
> <https://beta.golang.org/doc/go1.8>

---

> As close to how you would normally structure an app is best.

I'll look into this.